### PR TITLE
fix(script): Utility script to recompute clearing state of release

### PR DESCRIPTION
**Pull request by JaideepPalit**
> Utility script to recompute clearing state of release.
> * Which issue is this pull request belonging to and how is it solving it? (*#1127*)
> * Did you add or update any new dependencies that are required for your change? - No

> Have you implemented any additional tests? - No

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

File: scripts/utilities/004_recompute_clearing_state_of_release.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+couch = couchdb.Server(COUCHSERVER)
+db = couch[DBNAME]
+
+# ----------------------------------------
+# queries
+# ----------------------------------------
+
+# get all releases with attachment
+all_releases_with_attachments_query = {"selector": {"type": {"$eq": "release"}},"limit": 200000}
```
Comment (smrutis1) [inline scripts/utilities/004_recompute_clearing_state_of_release.py:44]: There seems to be no check for attachment in this query. Please add it or rename the variable.

File: scripts/utilities/004_recompute_clearing_state_of_release.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import json
+from webbrowser import get
+
+# ---------------------------------------
+# constants
+# ---------------------------------------
+
+DRY_RUN = True
+
+COUCHSERVER = "http://admin:root@localhost:5984/"
```
Comment (smrutis1) [inline scripts/utilities/004_recompute_clearing_state_of_release.py:33]: It would be good if username and password can be declared in separate vars.

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Not sure how the clearing state `SENT_TO_CLEARING_TOOL` and `UNDER_CLEARING` is handled.?

File: scripts/utilities/004_recompute_clearing_state_of_release.py (hunk 2 above)
Comment (JaideepPalit) [reply scripts/utilities/004_recompute_clearing_state_of_release.py]: Done

File: scripts/utilities/004_recompute_clearing_state_of_release.py (hunk 1 above)
Comment (JaideepPalit) [reply scripts/utilities/004_recompute_clearing_state_of_release.py]: Done

**Comment by JaideepPalit** [comment]
Thanks for review and test. Incorporated changes as suggested.

> Not sure how the clearing state SENT_TO_CLEARING_TOOL and UNDER_CLEARING is handled.?
   - Handled it.

New Logic to recompute Clearing state

- if release contains any attachment of type == `COMPONENT_LICENSE_INFO_XML` or `CLEARING_REPORT`
   - if `checkStatus` == `ACCEPTED` (attachment is accepted) -> set `clearing state` as `APPROVED`
   - else set clearing state as `REPORT_AVAILABLE`
- else if release `doesnot` contains any attachment of type == `COMPONENT_LICENSE_INFO_XML` or `CLEARING_REPORT`
   - check if current clearing state == `SENT_TO_CLEARING_TOOL` or `UNDER_CLEARING` which means it is a part of `FOSSology process` and will be updated on next iteration of `FOSSology process` , in such case ignore and move forward
   - else set `clearing state` as `NEW_CLEARING`

**Comment by mcjaeger** [comment]
has been tested on stage and productive

**Review by smrutis1 — APPROVED** [review]
Code looks good

**Pull request merged**